### ISSUE TITLE
Random Number node always triggers subsequent nodes

### ISSUE THREAD

**Issue by Pyrosopher** [WASasquatch/was-node-suite-comfyui#166]
When set to fixed, the Random Number node still triggers next nodes even though there hasn't been a change.

**Comment by WASasquatch** [comment]
Well it should work, what is the error? Maybe I forgot to update it for SDXL pooled output.

However, you just right click any CLIPTextEncode and **"Convert text to input"**

**Comment by WASasquatch** [comment]
Yeah the `IS_CHANGED` method is really weird to work with. Instead of just `True` it is changed, or `False` it isn't, it expects a hash from some data.

**Comment by everdrone** [comment]
Experiencing this problem right now, it's retriggering a huge amount of downstream nodes.
Any way to circumvent the problem?

**Comment by brbbbq** [comment]
I'm also running into this issue. I'm trying to incorporate the Random Number node into a workflow that generates a random crop. I'm noticing it triggers a render even if I'm just moving a node around and nothing else has changed.

https://github.com/user-attachments/assets/c124cccb-ec9d-466a-94dc-c1518bf8558e

**Comment by brbbbq** [comment]
In `WAS_Node_Suite.py`, I commented out lines `11942-11946`, and it seems to do the trick for the time being:
```
#    @classmethod
#    def IS_CHANGED(cls, seed, **kwargs):
#        m = hashlib.sha256()
#        m.update(seed)
#        return m.digest().hex()
```

**Comment by brbbbq** [comment]
In `WAS_Node_Suite.py`, under the `WAS_Random_Number` class, I commented out these lines, which seems to fix my situation for the time being:
```
#    @classmethod
#    def IS_CHANGED(cls, seed, **kwargs):
#        m = hashlib.sha256()
#        m.update(seed)
#        return m.digest().hex()
```

**Comment by WASasquatch** [comment]
Has there been a change to on change method? Hex for same seed should be the same, this match cached value.

**Comment by brbbbq** [comment]
> Has there been a change to on change method? Hex for same seed should be the same, thus match cached value.

I have no idea. I tried the WAS Seed node, and it works without issue. When I put it in fixed mode it doesn't render if there's no change.

Here's my version info:
ComfyUI: 2340[2dc84d](2024-07-06)
Manager: V2.46

**Comment by WASasquatch** [comment]
I'll have to look into it. The reason I have that in there is because I
don't rely on front end mechanics, so headless server stuff can still
function.

**Comment by brbbbq** [comment]
I really don't know anything about the code. I will say that it's not that the server isn't refreshing after every change, such as moving nodes around. In the console window it says:
```
[rgthree] Using rgthree's optimized recursive execution.
Prompt executed in 0.02 seconds
```

It's that the Random Number nodes is triggering the KSampler to render that's the issue. I don't know if that's helpful at all.

**Comment by WASasquatch** [comment]
Yeah I understand what's happening, but it shouldn't. The change function
reduces the seed to a unique hex, basically like a seed itself. The same
seed should always yield the same hexadecimal value and tell ConfyUI there
has been no change. It could be the rgthree execution server change but not
sure.
